runners/spark/src/main/java/org/apache/beam/runners/spark/TestSparkRunner.java

```diff
@@ -65,11 +77,23 @@ public static TestSparkRunner fromOptions(PipelineOptions options) {
     return new TestSparkRunner(sparkOptions);
   }
 
+  /**
+   * Overrides for the test runner.
+   */
+  @SuppressWarnings("unchecked")
   @Override
-  public <OutputT extends POutput, InputT extends PInput>
-      OutputT apply(PTransform<InputT, OutputT> transform, InputT input) {
-    return delegate.apply(transform, input);
-  };
+  public <OutputT extends POutput, InputT extends PInput> OutputT apply(
+          PTransform<InputT, OutputT> transform, InputT input) {
+    // if the pipeline is required to execute as a streaming pipeline,
+    // and the source is an adapted unbounded source (as bounded),
+    // read it as unbounded source via UnboundedReadFromBoundedSource.
+    if (isStreaming && transform instanceof BoundedReadFromUnboundedSource) {
```

Comment:
Can this be done in the translator? I know it is isolated to testing, but perhaps it can be a somewhat secret flag that you enable, like `delegate.setForceStreaming(true)`?

The reason I say this is that we actually want to eliminate `PipelineRunner.apply` so that the user completely constructs their pipeline before the runner has a handle on it.